### DISCUSSION TITLE
MemoryError: Process terminated: you likely have a memory-related error. Try increasing your Docker memory limit. only occurs when trying to run the pixel_cluster_utils.command

### DISCUSSION THREAD

**Discussion by whitneyt1** [angelolab/ark-analysis#731] [Q&A] [answered]
Hello! 

Dataset: 11 FOVs, each FOV is  400 x 400 

when running both `pixel_cluster_utils.cluster_pixels()` and `pixel_cluster_utils.pixel_consensus_cluster()` i receive the error `MemoryError: Process terminated: you likely have a memory-related error. Try increasing your Docker memory limit.`

After following the instructions and increasing my memory limit on Docker > Preferences > Resources > Memory, i still received the error. Additionally, I tried to address each specific error above the red error box. 

However, when I copied the code directly into jupyter notebooks, I was able to run the code successfully. Is there advice on how to prevent the MemoryError? Is there any reason I should he hesitant about using my files outputted via workaround? 

Error Cluster Pixels:
<img width="1197" alt="Screen Shot 2022-09-27 at 6 23 40 PM" src="https://user-images.githubusercontent.com/46573884/192666199-f0ead3e6-37ae-419a-829f-ce56e00c695a.png">
Attempt at Addressing Cluster Pixels Error:

<img width="948" alt="Screen Shot 2022-09-27 at 4 19 00 PM" src="https://user-images.githubusercontent.com/46573884/192654120-b069de19-eef2-4c4e-abaf-f2418656f1f9.png">

Workaround:
<img width="302" alt="Screen Shot 2022-09-27 at 4 23 43 PM" src="https://user-images.githubusercontent.com/46573884/192654555-bf82f3c1-ac89-49ab-b187-1b3178473f5b.png">

Error Pixel Consensus Cluster:
<img width="1034" alt="Screen Shot 2022-09-27 at 2 43 57 PM" src="https://user-images.githubusercontent.com/46573884/192641311-4e07dd5b-66de-4e80-b42b-d4388a7bd33a.png">

After setting seed manually:
<img width="939" alt="Screen Shot 2022-09-27 at 4 22 05 PM" src="https://user-images.githubusercontent.com/46573884/192654389-cb446697-9280-4ccf-80cf-ce407d3e21e9.png">

Work Around:
<img width="634" alt="Screen Shot 2022-09-27 at 2 44 14 PM" src="https://user-images.githubusercontent.com/46573884/192641354-59cf5065-dede-469c-96a7-0392adeccf38.png">

Output Directory: 
<img width="1013" alt="Screen Shot 2022-09-27 at 2 45 05 PM" src="https://user-images.githubusercontent.com/46573884/192641480-a3c1f161-dbae-4325-ba40-8b044aab3822.png">

**Comment by whitneyt1** [comment]
Similar problem when running `cell_cluster_utils.cell_consensus_cluster()` although I'm not able to perform an equivalent workaround successfully. This is with a modified cell table that removes all cell_table['label']==0

<img width="1431" alt="Screen Shot 2022-09-27 at 6 39 18 PM" src="https://user-images.githubusercontent.com/46573884/192667973-46be4a1c-efad-4bc3-97d5-0d8040375ca2.png">

**Reply by ngreenwald** [reply]
It says memory error at the bottom, but the actual error text at the top says you have invalid values when calculating the standard deviation. We should have the cell table fixed this week, once we do I think the easiest thing will be to re-run the whole pipeline to make sure there aren't any other hanging errors

**Reply by whitneyt1** [reply]
hi @ngreenwald, I saw there were some updates to ark, with the 0.4.2 release. Did this fix make it into that update or should i keep holding on? 

Thanks!

**Reply by ngreenwald** [reply]
We fixed one thing, but broke something else. Once 0.4.3 comes out it should all be resolved

**Comment by ngreenwald** [comment, marked as answer]
We've just made a new release that addresses this issue. Once you [follow the instructions](https://github.com/angelolab/ark-analysis#updating-the-repository) for updating, this should be resolved. I would recommend starting from the beginning of the whole pipeline in a clean folder to make sure you're using only updated files for all steps.
